statsmodels/tsa/statespace/dynamic_regression.py

```diff
@@ -0,0 +1,321 @@
+            if mod['local_level'] and not mod['trend']:
+                t = np.ones((1,1))
+            else:
+                t = np.array([[1,1],[0,1]])
+            if mod['irregular']:
+                t = block_diag(np.zeros((1,1)),t)
+            transition[start:end,start:end] = t
+            start = end
+        return transition
+
+    
+    @property
+    def initial_selection(self):
+        """Initial selection matrix"""
+        selection = np.zeros((self.k_states,self.k_state_cov))
+        start = 0
+        for mod in self.exog_models:
+            k_components = 0
+            if mod['local_level'] != "deterministic":
+                s = np.ones((1,1))
+                k_components += 1
+            if mod['trend'] & (mod['trend']!="deterministic"):
+                s = block_diag(s,np.ones((1,1)))
+                k_components += 1
+            if mod['irregular']:
+                s = block_diag(np.zeros((1,1)),s)
+                k_components += 1
+            end = start + k_components
+            selection[start:end,start:end] = s
+        return selection
+
+    def update(self, params, transformed=True, **kwargs):
+        """
+        Update the parameters of the model
+
+        Updates the representation matrices to fill in the new parameter
+        values.
+
+        Parameters
+        ----------
+        params : array_like
+            Array of new parameters.
+        transformed : boolean, optional
+            Whether or not `params` is already transformed. If set to False,
+            `transform_params` is called. Default is True..
+
+        Returns
+        -------
+        params : array_like
+            Array of parameters.
+        """
+        obs_cov = params[0]
+        state_cov_params = params[1:]
+        self.ssm.obs_cov[0,0,0]= obs_cov
+        self.ssm.state_cov[:,:,0][np.diag_indices(
+            self.k_state_cov)] = state_cov_params
+        
+
+    def transform_params(self, unconstrained):
+        """
+        Transform unconstrained parameters used by the optimizer to constrained
+        parameters used in likelihood evaluation.
+
+        Used primarily to enforce stationarity of the autoregressive lag
+        polynomial, invertibility of the moving average lag polynomial, and
+        positive variance parameters.
+
+        Parameters
+        ----------
+        unconstrained : array_like
+            Unconstrained parameters used by the optimizer.
+
+        Returns
+        -------
+        constrained : array_like
+            Constrained parameters used in likelihood evaluation.
+        """
+        unconstrained = np.array(unconstrained, ndmin=1)
+        constrained = np.zeros(unconstrained.shape, unconstrained.dtype)
+        start = 0
+        # Transform the covariance params to be positive
+        constrained[start:] = unconstrained[start:]**2
+        return constrained
+    
+    def untransform_params(self, constrained):
+        """
```

Comment:
Are these useful publically or should they be private?

Reply:
`transform_params` and `untransform_params` are part of the typical state space class structure, so they're public (for better or worse)